src/github.com/getlantern/flashlight/flashlight.go

```diff
@@ -229,7 +225,16 @@ func runClientProxy(cfg *config.Config) {
 
 	go func() {
 		addExitFunc(pacOff)
-		client.ListenAndServe(pacOn)
+		client.ListenAndServe(func() {
+			pacOn()
+			if showui {
+				// Launch a browser window with Lantern but only after the pac
+				// URL and the proxy server are all up and running to avoid
+				// race conditions where we change the proxy setup while the
+				// UI server and proxy server are still coming up.
+				ui.Show()
+			}
+		})
```

Comment:
The ``ListenAndServe`` callback will only be invoked if we successfully start accepting connections, but if we get an error, we really aren't accounting for it or logging it anywhere. Would be a good change to make while we're in here?